x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/tool/ResetElasticPasswordTool.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.tool;
+
+import joptsimple.OptionSet;
+
+import joptsimple.OptionSpecBuilder;
+
+import org.elasticsearch.cli.ExitCodes;
+import org.elasticsearch.cli.Terminal;
+import org.elasticsearch.cli.UserException;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.settings.KeyStoreWrapper;
+import org.elasticsearch.common.settings.SecureString;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.json.JsonXContent;
+import org.elasticsearch.core.CheckedFunction;
+import org.elasticsearch.env.Environment;
+import org.elasticsearch.xpack.core.security.support.Validation;
+
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.List;
+import java.util.function.Function;
+
+public class ResetElasticPasswordTool extends BaseRunAsSuperuserCommand {
+
+    private final Function<Environment, CommandLineHttpClient> clientFunction;
+    private final OptionSpecBuilder interactive;
+    private final OptionSpecBuilder auto;
+    private final OptionSpecBuilder batch;
+
+    ResetElasticPasswordTool() {
+        this(environment -> new CommandLineHttpClient(environment), environment -> KeyStoreWrapper.load(environment.configFile()));
+    }
+
+    public static void main(String[] args) throws Exception {
+        exit(new ResetElasticPasswordTool().main(args, Terminal.DEFAULT));
+    }
+
+    protected ResetElasticPasswordTool(
+        Function<Environment, CommandLineHttpClient> clientFunction,
+        CheckedFunction<Environment, KeyStoreWrapper, Exception> keyStoreFunction) {
+        super(clientFunction, keyStoreFunction, "Resets the password or the elastic built-in user");
```

Comment:
or => of
Also, I believe the description is supposed to be the help message

Reply:
> Also, I believe the description is supposed to be the help message

It is !